src/lib/y2users/linux/users_writer.rb

```diff
@@ -0,0 +1,366 @@
+
+      # Writes authorized keys for given user
+      #
+      # @see Yast::Users::SSHAuthorizedKeyring#write_keys
+      #
+      # @param user [User]
+      # @param issues [Y2Issues::List] a collection for adding issues if something goes wrong
+      def write_auth_keys(user, issues)
+        return unless user.home
+
+        Yast::Users::SSHAuthorizedKeyring.new(user.home, user.authorized_keys).write_keys
+      rescue Yast::Users::SSHAuthorizedKeyring::PathError => e
+        issues << Y2Issues::Issue.new(
+          # TRANSLATORS: %s is a placeholder for a username
+          format(_("Error writing authorized keys for '%s'"), user.name)
+        )
+        log.error("Error writing authorized keys for '#{user.name}' - #{e.message}")
+      end
+
+      # Executes the command for setting the password of given user
+      #
+      # @param user [User]
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def set_password_value(user, issues)
+        return unless user.password&.value
+
+        Yast::Execute.on_target!(CHPASSWD, *chpasswd_options(user))
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          # TRANSLATORS: %s is a placeholder for a username
+          format(_("The password for '%s' could not be set"), user.name)
+        )
+        log.error("Error setting password for '#{user.name}' - #{e.message}")
+      end
+
+      # Executes the command for setting the dates and limits in /etc/shadow
+      #
+      # @param user [User]
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def set_password_attributes(user, issues)
+        return unless user.password
+
+        options = chage_options(user)
+
+        return if options.empty?
+
+        Yast::Execute.on_target!(CHAGE, *options, user.name)
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          # TRANSLATORS: %s is a placeholder for a username
+          format(_("Error setting the properties of the password for '%s'"), user.name)
+        )
+        log.error("Error setting password attributes for '#{user.name}' - #{e.message}")
+      end
+
+      # Attributes to modify using `usermod`
+      USERMOD_ATTRS = [:gid, :home, :shell, :gecos].freeze
+
+      # Edits the user
+      #
+      # @param new_user [User] User containing the updated information
+      # @param old_user [User] Original user
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def edit_user(new_user, old_user, issues)
+        usermod_changes = USERMOD_ATTRS.any? do |attr|
+          !new_user.public_send(attr).nil? &&
+            (new_user.public_send(attr) != old_user.public_send(attr))
+        end
+        usermod_changes ||= different_groups?(new_user, old_user)
+
+        Yast::Execute.on_target!(USERMOD, *usermod_options(new_user, old_user)) if usermod_changes
```

Comment:
and same what I mention above also here. Here it is even more visible as it contain also usermod detection.

Reply:
This is again a re-review of old code that was just moved. Let's keep that out for the time being.